pinot-common/src/main/java/org/apache/pinot/segment/local/customobject/DoubleVector.java

```diff
@@ -0,0 +1,106 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.segment.local.customobject;
```

Comment:
What's the alternative to moving this from pinot-segment-local module to pinot-common ?

It seems fine to me and I wonder why `customobject` was in pinot-segment-local module in the first place. It is generated by server during execution and then used  in `DataSchema`, `DataTable`, `ObjectSerDeUtils` and Reduce code all of which are in either pinot-common or pinot-core. So `customobject` should ideally be in pinot-core / pinot-common imo.

cc @richardstartin  @Jackie-Jiang for any thoughts on why this should be in pinot-segment-local ?

Reply:
We do have `it.unimi.dsi.fastutil.doubles.DoubleArrayList` already supported that can be used for ser-de. Then we don't need to move these classes.

These objects are for ser/de of the intermediate results, instead of the final results, which will be used to construct the final result table.

Reply:
Yes we can use `DoubleArrayList` for the final result, which will not require moving customobjects. But for intermediate result I will keep the `DoubleVector` since `DoubleArrayList` lacks the basic vector operators, e.g., vector add, dot product, etc. These operators are useful for the implementation of statistical functions.

Reply:
I'd suggest also using `DoubleArrayList` for the intermediate result, and add some helper functions in the aggregation function implementation (or a separate util class if you think these are common operations and can be shared across other functions added in the future) to do vector add etc on top of the `DoubleArrayList`.
Both of these 2 classes are just Comparable wrapper on top of the double array. We probably don't want to maintain 2 separate custom objects for the same functionality.

Reply:
Sure. Moved the functions to a util class.